prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java

```diff
@@ -273,29 +275,67 @@ public APIResult resume(@PathParam("job-name") String jobName,
     @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
     @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
     public APIResult delete(@PathParam("job-name") String jobName,
+                            @Context HttpServletRequest request,
                             @DefaultValue("") @QueryParam("doAs") String doAsUser) {
         checkIfExtensionServiceIsEnabled();
+        ExtensionMetaStore metaStore = ExtensionStore.getMetaStore();
+        ExtensionJobsBean extensionJobDetails = metaStore.getExtensionJobDetails(jobName);
+        if (extensionJobDetails == null) {
+            // return failure if the extension job doesn't exist
+            return new APIResult(APIResult.Status.SUCCEEDED,
+                    "Extension job " + jobName + " doesn't exist. Nothing to delete.");
+        }
+
+        SortedMap<EntityType, List<Entity>> entityMap;
         try {
+                entityMap = getJobEntities(jobName, extensionJobDetails, doAsUser);
+                deleteEntities(entityMap, request);
+        } catch (FalconException | IOException | JAXBException e) {
+            LOG.error("Error when deleting extension job: " + jobName + ": ", e);
+            throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+        metaStore.deleteExtensionJob(jobName);
+        return new APIResult(APIResult.Status.SUCCEEDED, "Extension job " + jobName + " deleted successfully");
+    }
+
+    private SortedMap<EntityType,List<Entity>> getJobEntities(String jobName, ExtensionJobsBean extensionJobsBean,
+                                                              String doAsUser) throws FalconException, IOException {
+        TreeMap<EntityType, List<Entity>> entityMap = new TreeMap<>();
+        ExtensionMetaStore metaStore = ExtensionStore.getMetaStore();
+
+        ExtensionType extensionType =  metaStore.getDetail(extensionJobsBean.getExtensionName()).getExtensionType();
+        if (ExtensionType.TRUSTED.equals(extensionType)) {
```

Comment:
Why do this? The list of processes and feeds are stored in the Jobs table, even for trusted extensions.